WORKSPACE

```diff
@@ -80,77 +80,77 @@ maven_jar(
     sha1 = "cce0823396aa693798f8882e64213b1772032b09",
 )
 
-# For our scala_image test.
-git_repository(
-    name = "io_bazel_rules_scala",
-    commit = "0bac7fe86fdde1cfba3bb2c8a04de5e12de47bcd",
-    remote = "https://github.com/bazelbuild/rules_scala.git",
-)
+LANGUAGES = [
+    {
+        "name": "scala",
+        "commit": "0bac7fe86fdde1cfba3bb2c8a04de5e12de47bcd",
+        "sha256": "f5f35de94d2d64e48fb4aef87cf89248b8980cb25f9ff1449575af8d904f41be",
+    },
+    {
+        "name": "groovy",
+        "commit": "6b8e32ce0f7e33ae1b859706c2dc0c169b966e7e",
+        "sha256": "ced0b80f4c32805abc61592eb2e5bd3e1ab809abdc24e39f888ce33055bfee5a",
+    },
+    {
+        "name": "go",
+        "commit": "74849e006b4f4392e95a12188cd0352b01730c3a",
+        "sha256": "ac1ba9a8527ccb476440cb5eb754849ed3fcd9a0c5103d9ced047fa33ffcfffe",
+    },
+    {
+        "name": "rust",
+        "commit": "7b1ba1f2a89006fbe358e97011cb1c1516435806",
+        "sha256": "69ce87d3697a9f2f092b4a07be2a85dbd472188173726702ca55c6e51d2ca0c9",
+    },
+    {
+        "name": "d",
+        "commit": "0400b9b054013274cee2ed15679da19e1fc94e07",
+        "sha256": "527908e02d7bccf5a4eb89b690b003247eb6c57d69cc3234977c034d27c59d6e",
+    },
+]
+
+[http_archive(
+    name = "io_bazel_rules_{}".format(l["name"]),
+    sha256 = l["sha256"],
+    url = "https://github.com/bazelbuild/rules_{}/archive/{}.tar.gz".format(l["name"], l["commit"]),
+    strip_prefix = "rules_{}-{}".format(l["name"], l["commit"]),
+) for l in LANGUAGES]
```

Comment:
I have mixed feelings about the loop vs. just unrolling it.  I think it ultimately ends up being roughly the same number of lines, but takes a bit more parsing to know exactly what's going on.

I understand that `http_archive` can result in better build performance than `git_repository`, but I have seen at least one instance where things broke because GitHub changed something about the way they produce these archives.  I'm also not a particularly big fan of having to juggle two SHAs instead of one.

I think these are distractions from the core change you are after.  I'm happy to discuss the benefits of `http_archive`, but perhaps you could limit this first change to the `rules_go` and `rules_nodejs` updates?